torch/csrc/jit/passes/quantization/insert_quant_dequant.cpp

```diff
@@ -745,6 +866,55 @@ void propagateQuantizationOps(Block* block) {
   }
 }
 
+void InsertQuantDeQuantHelper::runWeightObserver(
+    Module& module,
+    const std::string& method_name) {
+  for (auto& invoked_methods : getInvokedMethods(module, method_name)) {
+    auto& invoked_module = std::get<0>(invoked_methods);
+    const auto& invoked_method_name = std::get<1>(invoked_methods);
+    runWeightObserver(invoked_module, invoked_method_name);
+  }
+  Method method = module.get_method(method_name);
+  auto graph = method.graph();
+  Value* self = graph->inputs()[0];
+
+  std::vector<Value*> weight_values;
+  for (size_t idx = 1; idx < method.num_inputs(); ++idx) {
+    auto& v = graph->inputs()[idx];
+    if (v->type()->isSubtypeOf(TensorType::get())) {
+      auto observer_name = findObserverName(v);
+      if (observer_name && isWeight(module, v)) {
+        weight_values.push_back(v);
+      }
+    }
+  }
+  std::stack<Block*> blocks_to_visit;
```

Comment:
For understanding, at this point, we have a stack of weight tensors in weight_values. What does the code below do? It seems to be finding other weights? Can you add a comment clarifying the logic?

Reply:
It traverses all the blocks in the graph to get the observed weight values.